lib/Doctrine/ORM/Cache/DefaultCacheFactory.php

```diff
@@ -178,6 +181,19 @@ public function buildQueryCache(EntityManagerInterface $em, $regionName = null)
      */
     public function buildCollectionHydrator(EntityManagerInterface $em, array $mapping)
     {
+        if ($mapping['cache']) {
+            $targetPersister = $em->getUnitOfWork()->getEntityPersister($mapping['targetEntity']);
+
+            if ($targetPersister instanceof CachedPersister) {
+
+                $targetRegion = $targetPersister->getCacheRegion();
+
+                if ($targetRegion instanceof MultiGetRegion) {
+                    return new MultiGetCollectionHydrator($em);
```

Comment:
The multi-get region is not being passed to the `MultiGetCollectionHydrator`? You are doing a lot of checks but discarding a lot of dependencies then: is it intended to be like that?